packages/shared/enqueueTask.js

```diff
@@ -8,37 +8,41 @@
  */
 
 let didWarnAboutMessageChannel = false;
-let enqueueTask;
-try {
-  // read require off the module object to get around the bundlers.
-  // we don't want them to detect a require and bundle a Node polyfill.
-  let requireString = ('require' + Math.random()).slice(0, 7);
-  let nodeRequire = module && module[requireString];
-  // assuming we're in node, let's try to get node's
-  // version of setImmediate, bypassing fake timers if any.
-  enqueueTask = nodeRequire('timers').setImmediate;
-} catch (_err) {
-  // we're in a browser
-  // we can't use regular timers because they may still be faked
-  // so we try MessageChannel+postMessage instead
-  enqueueTask = function(callback: () => void) {
-    if (__DEV__) {
-      if (didWarnAboutMessageChannel === false) {
-        didWarnAboutMessageChannel = true;
-        if (typeof MessageChannel === 'undefined') {
-          console.error(
-            'This browser does not have a MessageChannel implementation, ' +
-              'so enqueuing tasks via await act(async () => ...) will fail. ' +
-              'Please file an issue at https://github.com/facebook/react/issues ' +
-              'if you encounter this warning.',
-          );
+let enqueueTaskImpl = null;
```

Comment:
this had to be rewritten to prevent some unneeded code from leaking into the prod build 
![image](https://user-images.githubusercontent.com/18808/73267935-e722a880-41d1-11ea-9049-e7df25f73657.png)

Reply:
Nit: Use the feature flag instead?

```js
function enqueueTask(task: () => void) {
  if (isTestingEnvironment) {
    // ...
  }
}
```

Reply:
ooh, that could work.

Reply:
actually, I'll wait on this for the next PR, else I'll have to include changing all our tests to using the test build in this PR.